source/Preprocessor.cpp

```diff
@@ -938,7 +938,12 @@ MacroActualArgumentSyntax* Preprocessor::MacroParser::parseActualArgument() {
 }
 
 MacroFormalArgumentSyntax* Preprocessor::MacroParser::parseFormalArgument() {
-    auto arg = expect(TokenKind::Identifier);
+    Token arg = pp.peek(currentMode);
```

Comment:
You should probably use peek() here instead of pp.peek()... see the function definition below this. It does some other stuff like looking in the local macro buffer.